src/test/ui/pattern/pat-tuple-underfield.stderr

```diff
@@ -40,13 +40,13 @@ LL |         S(..) => {}
    |           ~~
 
 error[E0023]: this pattern has 0 fields, but the corresponding tuple struct has 2 fields
-  --> $DIR/pat-tuple-underfield.rs:20:9
+  --> $DIR/pat-tuple-underfield.rs:20:10
    |
 LL | struct S(i32, f32);
-   | ------------------- tuple struct defined here
+   |          ---  --- tuple struct has 2 fields
 ...
 LL |         S() => {}
-   |         ^^^ expected 2 fields, found 0
+   |         -^^ expected 2 fields, found 0
```

Comment:
The spans might look a little fun in these cases if the user is using non-standard formatting. I think it would like this:

```
LL |         S () => {}
   |         -^^^ expected 2 fields, found 0
```

In case of no fields, should we just use `^` for the whole pattern? (Perhaps that might be a solution to https://github.com/rust-lang/rust/pull/88123#r693390008 as well.)

Reply:
Yeah, that'd be reasonable. We do similar things elsewhere (special case the 0-case to point at the whole thing to make it easier to read).